sdk/communication/communication-common/test/public/node/entraCommunicationTokenCredential.spec.ts

```diff
@@ -58,6 +63,136 @@ describe("Entra CommunicationTokenCredential", function () {
+      resourceEndpoint: resourceEndpoint,
+      tokenCredential: tokenCredential,
+      scopes,
+    };
+    const entraTokenCredential = new EntraTokenCredential(entraTokenCredentialOptions);
+    const exchangeTokenSpy = sinon.spy(entraTokenCredential as any, "exchangeEntraToken");
+
+    let tokenResult = (await entraTokenCredential.getToken()).token;
+    assert.strictEqual(tokenResult, acsToken);
+    assert.isTrue(exchangeTokenSpy.callCount > 0);
+    assert.isTrue(scope.isDone());
+
+    scope = successApiMock();
+    exchangeTokenSpy.restore();
+    exchangeTokenSpy.resetHistory();
+
+    tokenResult = (await entraTokenCredential.getToken()).token;
+    assert.strictEqual(tokenResult, acsToken);
+    assert.isTrue(exchangeTokenSpy.callCount === 0);
+    exchangeTokenSpy.restore();
+  });
+
+  it("Token exchange gets called again when Entra token changes", async function () {
+    let scope = successApiMock();
+    let entraTokenCredentialOptions: EntraCommunicationTokenCredentialOptions = {
+      resourceEndpoint: resourceEndpoint,
+      tokenCredential: tokenCredential,
+      scopes,
+    };
+    let entraTokenCredential = new EntraTokenCredential(entraTokenCredentialOptions);
+    let exchangeTokenSpy = sinon.spy(entraTokenCredential as any, "exchangeEntraToken");
+
+    let tokenResult = (await entraTokenCredential.getToken()).token;
+    assert.strictEqual(tokenResult, acsToken);
+    assert.isTrue(exchangeTokenSpy.callCount > 0);
+    assert.isTrue(scope.isDone());
+
+    const apiMockNewToken = () =>
+      nock(resourceEndpoint)
+        .post(tokenExchangePath)
+        .matchHeader("Authorization", `Bearer ${newEntraToken}`);
+
+    const successApiMockNewToken = () =>
+      apiMockNewToken().reply(200, {
+        accessToken: {
+          token: acsToken,
+          expiresOn: new Date(Date.now() + 60 * 60 * 1000).toISOString(),
+        },
+      });
+
+    scope = successApiMockNewToken();
+    const acsTokenCredential: TokenCredential = {
+      getToken: async (_scopes: string, _options?: GetTokenOptions) => {
+        return {
+          token: newEntraToken,
+          expiresOnTimestamp: Date.now() + 60 * 60 * 1000,
+        };
+      },
+    };
+    entraTokenCredentialOptions = {
+      resourceEndpoint: resourceEndpoint,
+      tokenCredential: acsTokenCredential,
+      scopes,
+    };
+    entraTokenCredential = new EntraTokenCredential(entraTokenCredentialOptions);
+    exchangeTokenSpy = sinon.spy(entraTokenCredential as any, "exchangeEntraToken");
+
+    tokenResult = (await entraTokenCredential.getToken()).token;
+    assert.strictEqual(tokenResult, acsToken);
+    assert.isTrue(exchangeTokenSpy.callCount > 0);
+    exchangeTokenSpy.restore();
+  });
+
+  it("Token exchange gets called again when acs token expires", async function () {
+    const currentDateTime = new Date(Date.now());
+    currentDateTime.setHours(currentDateTime.getHours() - 1);
```

Comment:
The name current is misleading, and this can be simplified a bit:

```suggestion
    const expiredDate = new Date(Date.now() - 1000 * 60);
```